### PR TITLE
FS-93 Changes for the File Upload UI

### PR DESCRIPTION
**Pull request by gaganahluwalia**
## Description
These changes are to do the UI changes for File Upload, backend is already done.

## Changelog
- New Components for the File Upload Part
- New CSS Files
- Changes to existing input.tsx file

### CODE REVIEW

File: frontend/src/components/fileUploader.module.css [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  width: 24px;
+  height: 24px;
+}
+
+.uploadButton:not(:disabled):hover {
+  cursor: pointer;
+}
+
+.uploadButton:active {
+  background-color: var(--grey-400);
```
Comment (dianaPrahoveanu-SL) [inline frontend/src/components/fileUploader.module.css:36]: I noticed when clicking the button the background has a square shape. I think that can be an easy fix by adding` border-radius: 50%;` here.

File: frontend/src/components/uploadedFileDisplay.module.css [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  align-items: center;
+  height: 36px;
+  margin-bottom: 4px;
+  box-sizing: border-box;
+}
+
+.uploadedFile {
+  font-size: 14px;
+  color: var(--grey-900);
+  background-color: var(--grey-100);
```
Comment (dianaPrahoveanu-SL) [inline frontend/src/components/uploadedFileDisplay.module.css:13]: I think the background colour here should be --grey-200

File: frontend/src/components/uploadedFileDisplay.module.css [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  padding: 8px 12px;
+  border-radius: 8px;
+  text-align: left;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  margin-right: 65px;
+  border: 1px solid var(--grey-300);
```
Comment (dianaPrahoveanu-SL) [inline frontend/src/components/uploadedFileDisplay.module.css:21]: Maybe a nit-pick but I don't think we should have a border on this card here as per the UX specs

File: frontend/src/components/uploadedFileDisplay.module.css [hunk 4]
```diff
@@ -0,0 +1,29 @@
+.uploadedFileContainer {
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  height: 36px;
+  margin-bottom: 4px;
+  box-sizing: border-box;
+}
+
```
Comment (dianaPrahoveanu-SL) [inline frontend/src/components/uploadedFileDisplay.module.css:9]: I noticed some extra spacing between the inputContainer and the uploadedFileContainer which I think can be easily fixed by removing the margin from the inputContainer

**Review by dianaPrahoveanu-SL — COMMENTED** [review]
Great work 💯 I just got a couple small comments

File: frontend/src/components/input.tsx [hunk 5]
```diff
@@ -8,7 +8,8 @@ import React, {
 } from 'react';
 import styles from './input.module.css';
 import RightArrowIcon from '../icons/send.svg';
-import UploadIcon from '../icons/upload.svg';
+import { FileUploader } from './fileUploader';
```
Comment (IMladjenovic) [inline frontend/src/components/input.tsx:11]: Can we call it `FileUpload` instead?

File: frontend/src/components/fileUploader.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import React, { ChangeEvent, useState } from 'react';
+import styles from './fileUploader.module.css';
+import UploadIcon from '../icons/upload.svg';
+import UploadInProgressIcon from '../icons/upload-in-progress.svg';
+import CheckCircleIcon from '../icons/check-circle.svg';
+
+interface FileUploaderProps {
+  onFileUpload: (file: File) => Promise<void>;
+  isUploading: boolean;
+  isUploaded: boolean;
```
Comment (IMladjenovic) [inline frontend/src/components/fileUploader.tsx:10]:
```suggestion
  uploadInProgress: boolean;
  uploadComplete: boolean;
```
I would suggest these names are slightly nicer to read

File: frontend/src/components/input.tsx [hunk 7]
```diff
@@ -20,6 +21,11 @@ export interface InputProps {
 
 export const Input = ({ sendMessage, waiting, suggestions }: InputProps) => {
   const [userInput, setUserInput] = useState<string>('');
+  const [uploadedFile, setUploadedFile] = useState<File | null>(null);
+  const [isUploading, setIsUploading] = useState<boolean>(false);
+  const [isUploaded, setIsUploaded] = useState<boolean>(false);
+  // TODO: Add fileId to the state
+  // const [fileId, setFileId] = useState<string>('');
```
Comment (IMladjenovic) [inline frontend/src/components/input.tsx:28]: Is this for future use?

**Review by IMladjenovic — CHANGES_REQUESTED** [review]

File: frontend/src/components/input.tsx [hunk 8]
```diff
@@ -52,26 +58,59 @@ export const Input = ({ sendMessage, waiting, suggestions }: InputProps) => {
...
+      formData.append('file', file);
+      const response = await fetch(`${process.env.BACKEND_URL}/uploadfile`, {
+        method: 'POST',
+        body: formData,
+        credentials: 'include',
+      });
+      if (!response.ok)
+        throw new Error(`Upload failed with status ${response.status}`);
+
+      const { filename, id } = await response.json();
```
Comment (mic-smith) [inline frontend/src/components/input.tsx:76]: Could we move this request and response bit to `server.ts` to match the other api calls?

File: frontend/src/components/input.tsx [hunk 9]
```diff
@@ -52,26 +58,59 @@ export const Input = ({ sendMessage, waiting, suggestions }: InputProps) => {
...
+      console.log(`File uploaded successfully: ${filename} with id ${id}`);
+      // TODO: Set fileId in the state
+      // setFileId(id);
+      setUploadedFile(file);
+      setIsUploading(false);
+      setIsUploaded(true);
```
Comment (mic-smith) [inline frontend/src/components/input.tsx:82]: Do we need the `isUploaded` state? We could maybe just use the fact the `uploadedFile` is set?

File: frontend/src/components/fileUploader.module.css (hunk 1 above)
Comment (gaganahluwalia) [reply frontend/src/components/fileUploader.module.css]: Done :)

File: frontend/src/components/uploadedFileDisplay.module.css (hunk 2 above)
Comment (gaganahluwalia) [reply frontend/src/components/uploadedFileDisplay.module.css]: Done :)

File: frontend/src/components/uploadedFileDisplay.module.css (hunk 3 above)
Comment (gaganahluwalia) [reply frontend/src/components/uploadedFileDisplay.module.css]: Done :)

File: frontend/src/components/uploadedFileDisplay.module.css (hunk 4 above)
Comment (gaganahluwalia) [reply frontend/src/components/uploadedFileDisplay.module.css:9]: Done :)

File: frontend/src/components/input.tsx (hunk 5 above)
Comment (gaganahluwalia) [reply frontend/src/components/input.tsx]: Done :)

File: frontend/src/components/fileUploader.tsx (hunk 6 above)
Comment (gaganahluwalia) [reply frontend/src/components/fileUploader.tsx]: Done :)

File: frontend/src/components/input.tsx (hunk 7 above)
Comment (gaganahluwalia) [reply frontend/src/components/input.tsx]: Yes, if we needed fileId, but there are other ways and depends upon the design to get that report, so I have removed it.

File: frontend/src/components/input.tsx (hunk 8 above)
Comment (gaganahluwalia) [reply frontend/src/components/input.tsx]: Done :)

File: frontend/src/components/input.tsx (hunk 9 above)
Comment (gaganahluwalia) [reply frontend/src/components/input.tsx]: Yes, you are right. Initially I used it for quick work, but then forgot to use the uploadedFile and remove it.

**Review by IMladjenovic — APPROVED** [review]
Looks good. Haven't pulled it down since the changes and tested myself though

**Review by dianaPrahoveanu-SL — APPROVED** [review]

File: frontend/src/components/fileUpload.tsx [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  const handleFileChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file) {
+      onFileUpload(file);
+    }
+  };
+
+  const tooltipContent = disabled ? (
+    <>
+      <p>You already uploaded one file.</p>
```
Comment (mic-smith) [inline frontend/src/components/fileUpload.tsx:29]: This matches design, but should probably be `You've` instead of `You`.

**Review by mic-smith — APPROVED** [review]

**Pull request merged**